map-integration/macrostrat/map_integration/custom_integrations/gems_utils.py

```
@@ -22,35 +24,39 @@ def extract_gdb_layer(legend_path, layer_name, read_geometry):
     """
 
     dmu_layer = None
-
-    try:
-        # we are trying to read all of the non-spatial layers in the .gdb
-        # ingest any non-spatial metadata from .gdb just specify the file name in the re.search function
-        # ex. GeoMaterialDict or DescriptionOfMapUnits
-        # then we merge legend_df into the polygons df on a join_col
-        for name in fiona.listlayers(legend_path):
-            if re.search(rf"{layer_name}", name, re.IGNORECASE):
-                dmu_layer = name
-        if dmu_layer is None:
-            print(
-                f"[yellow]No {layer_name} table found in "
-                f"{legend_path.name}.  Layers: "
-                f"{', '.join(fiona.listlayers(legend_path))}[/yellow]"
-            )
-            return None
-        legend_df = G.read_file(
-            legend_path,
-            layer=dmu_layer,
-            engine="pyogrio",
-            read_geometry=read_geometry,
-        )
-        return legend_df
-    except ValueError as e:
-        print(f"[red]Error {e}[/red]\n")
```

could just allow the error to bubble up rather than returning none? This probably works fine; depends on how significant the error is